Mathlib/Data/Matrix/Rank.lean

```diff
@@ -274,4 +275,39 @@ theorem rank_eq_finrank_span_row [LinearOrderedField R] [Finite m] (A : Matrix m
   rw [← rank_transpose, rank_eq_finrank_span_cols, transpose_transpose]
 #align matrix.rank_eq_finrank_span_row Matrix.rank_eq_finrank_span_row
 
+section MulNonsingInv
+
+/-! ### Lemmas about pre or post multiplying by an invertible matrix. -/
+
+/-- Pre (Left) multiplying by an invertible matrix does not change the rank -/
+lemma rank_mul_IsUnit {m n R: Type}
```

Comment:
You should use `Type _` where possible so that your lemmas are applicable for types in higher universes

Reply:
are there matrices in higher universes?

Reply:
In principle yes, in practice not so much I don't think. Neverthess it's good style to be flexible just in case. Especially for a theoretical result like these, that could be useful in a more general context. Types in higher universes certainly do occur in mathlib

Reply:
At any rate, there should be existing variables at the top of the file that you can just re-use

Reply:
Ok I removed the variables in the declarations and reused the ones from the top of the file. I added the DecidableEq instances since these are needed for the definition of determinant.